### PR TITLE
workflow updates

### PR DESCRIPTION
**Pull request by 0xc0170**
Open for review

### CODE REVIEW

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -28,6 +28,14 @@ A member of the Mbed team must verify bug fixes before we pull the fixes into th
 
 The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit is merged into the default branch.
 
+#### Features
+
+Any feature should be implemented on the branch in mbed OS repository first.
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: New features should initially be implemented on a separate branch in the mbed OS repository. The naming convention for a feature branch is: "feature-<feature name>"

File: docs/reference/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -28,6 +28,14 @@ A member of the Mbed team must verify bug fixes before we pull the fixes into th
 
 The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit is merged into the default branch.
 
+#### Features
+
+Any feature should be implemented on the branch in mbed OS repository first. 
+
+Each feature has a tech lead. This person is responsible for:
+- rebased often to track master development
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: rebasing

File: docs/reference/contributing/guidelines/workflow.md [hunk 3]
```diff
@@ -28,6 +28,14 @@ A member of the Mbed team must verify bug fixes before we pull the fixes into th
...
+- shall review any addition to the feature branch (approval required by the feature owner or person assigned to do instead)
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: reviewing any additions

File: docs/reference/contributing/guidelines/workflow.md [hunk 4]
```diff
@@ -36,31 +44,35 @@ Please see the [code contributions chapter](/docs/v5.7/reference/guidelines.html
...
-	1. Use the imperative mood in the subject line.
-	1. Wrap the body at 72 characters.
-	1. Use the body to explain _what_ and _why_ vs _how_.
+    1. Separate subject from body with a blank line.
+    1. Limit the subject line to 72 characters (note that this is a deviation from Beam's standard).
+    1. Capitalize the subject line.
+    1. Do not end the subject line with a period.
+    1. Use the imperative mood in the subject line.
+    1. Wrap the body at 72 characters.
+    1. Use the body to explain _what_ and _why_ vs _how_.
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Why the spaces? The indentation should be handled automatically by the markdown tags. Though looks like you have lost those for some reason hence the '1.' everywhere ?

File: docs/reference/contributing/guidelines/workflow.md [hunk 5]
```diff
@@ -36,31 +44,35 @@ Please see the [code contributions chapter](/docs/v5.7/reference/guidelines.html
...
-3. Guide contributors both technically and procedurally.
-4. Run pull requests through the CI systems.
-5. Merge pull requests into the requested branches.
-6. Make periodic patch and feature releases.
+1. Ensure the relevant stakeholders review pull requests.
+1. Guide contributors both technically and procedurally.
+1. Run pull requests through the CI systems.
+1. Put label version.
+1. Merge pull requests into the requested branches.
+1. Make periodic patch and feature releases.
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Again you seem to have lost the numbering ?

File: docs/reference/contributing/guidelines/workflow.md [hunk 6]
```diff
@@ -69,6 +81,16 @@ The current maintainers are:
...
+* Cruz Monrreal (cmonr).
+
+#### Stakeholders
+
+They are responsible for a specific mbed OS module or feature.
+
+Responsibilities:
+
+1. Review pull request and coordinate reviews within the team
+1. Ensure the version label is correct
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Numbering ?

File: docs/reference/contributing/guidelines/workflow.md [hunk 7]
```diff
@@ -80,14 +102,28 @@ Each pull request goes through the following workflow:
 
 Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
 
+Each state is time boxed. In most cases, this should be sufficient time to move to an another state.
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md:107]: move to another state

File: docs/reference/contributing/guidelines/workflow.md [hunk 8]
```diff
@@ -80,14 +102,28 @@ Each pull request goes through the following workflow:
...
+
 ##### Reviews
 
-All pull requests must be reviewed. The Mbed OS maintainers determine the most suitable person to review the pull request and tag that person accordingly. If a pull request requires a review from partners, the maintainers tag the corresponding GitHub group instead.
+All pull requests must be reviewed. The mbed CI bot determine the most suitable person to review the pull request and tag that person accordingly.
+
+Time: 3 days
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Better to explain this specifically. E.g.
Is 3 days the maximum allowance for the initial review to take place, the full review cycle or for each review iteration ? We need to clarify this.

File: docs/reference/contributing/guidelines/workflow.md [hunk 9]
```diff
@@ -80,14 +102,28 @@ Each pull request goes through the following workflow:
...
 
 ##### The CI (Continuous Integration) testing
 
 There are a number of CI systems available. Which CI tests we run against a particular pull request depends on the effect of that pull request to the code base. Irrespective of which CIs tests run, Mbed OS has an all green policy, meaning that all the CI jobs that are triggered must pass before we merge the pull request.
 
+Time: 1 day
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Again, what do we mean by this? Are we saying this is the time limit between marking something as needs: ci and actually triggering it, or is it the time between triggering it and getting it to pass, or a combination of the two ?

File: docs/reference/contributing/guidelines/workflow.md [hunk 10]
```diff
@@ -80,14 +102,28 @@ Each pull request goes through the following workflow:
...
+
+The CI workflow can be found [here](tbd)
+
+##### Work needed
+
+The pull request requires additional work due to failed tests, changes requested.
+
+Time: 3 day
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Again clarification needed as to what this means

File: docs/reference/contributing/guidelines/workflow.md [hunk 11]
```diff
@@ -102,8 +138,6 @@ We use many other labels to summarize the scope and effect of the changes.
 
 - *needs: preceding PR* - This pull request cannot yet be merged because it has a dependency on another pull request that needs to merge first.
 - *DO NOT MERGE* - This pull request contains changes that may be in a draft state and submitted purely for review, or may contain breaking changes that have not been considered.
-- *devices: 'name'* - The pull request specifically affects the named device(s).
-- *component: 'name'* - The pull request specifically affects the named component.
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: Why have you removed these two labels ?

**Review by adbridge — CHANGES_REQUESTED** [review]

File: docs/reference/contributing/guidelines/workflow.md (hunk 4 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: yes will revert this somehow got messed during the updates

File: docs/reference/contributing/guidelines/workflow.md (hunk 5 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: markdown does it automatically, it is always a probelm when adding /changing - all numbers affected. this way can do without editing everything

File: docs/reference/contributing/guidelines/workflow.md (hunk 11 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: oh there are for issues, reverting :-)

File: docs/reference/contributing/guidelines/workflow.md [hunk 12]
```diff
@@ -12,31 +12,32 @@ Patch contributions can only be accepted through GitHub by creating a pull reque
...
+##### Fixing bugs
 
 A member of the Mbed team must verify bug fixes before we pull the fixes into the main branch. You must therefore use GitHub to fork the repo and then submit a pull request with your changes.
 
 The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit is merged into the default branch.
 
-#### Further reading
+##### Features
+
+New features should initially be implemented on a separate branch in the Mbed OS repository. The naming convention for a feature branch is: "feature-".
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: you've missed the <feature-name> git in the naming convention, we need to make this specific

File: docs/reference/contributing/guidelines/workflow.md (hunk 5 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Do your changes render correctly when you view with a markdown viewer now ?

File: docs/reference/contributing/guidelines/workflow.md [hunk 13]
```diff
@@ -80,14 +95,28 @@ Each pull request goes through the following workflow:
...
 
+Time: 1 day for CI to complete and report back results
+
+The CI workflow can be found [here](tbd)
+
+##### Work needed
+
+The pull request requires additional work due to failed tests, changes requested.
+
+Time: 3 day to respond to the review comments
```
Comment (adbridge) [inline docs/reference/contributing/guidelines/workflow.md]: days

**Review by adbridge — CHANGES_REQUESTED** [review]

File: docs/reference/contributing/guidelines/workflow.md (hunk 5 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: Yes, please view it to verify

File: docs/reference/contributing/guidelines/workflow.md (hunk 12 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: I do not follow what is missing?

Comment (cmonr) [reply docs/reference/contributing/guidelines/workflow.md]: I'm not sure I follow either.

File: docs/reference/contributing/guidelines/workflow.md [hunk 14]
```diff
@@ -12,31 +12,32 @@ Patch contributions can only be accepted through GitHub by creating a pull reque
...
 A member of the Mbed team must verify bug fixes before we pull the fixes into the main branch. You must therefore use GitHub to fork the repo and then submit a pull request with your changes.
 
 The last line in your commit message description should say “Fixes #deadbeef”, where “deadbeef” is the issue number in GitHub. This allows GitHub to automatically close the issue when the commit is merged into the default branch.
 
-#### Further reading
+##### Features
+
+New features should initially be implemented on a separate branch in the Mbed OS repository. The naming convention for a feature branch is: "feature-".
+
+Each feature has a tech lead. This person is responsible for:
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: Is there some mechanism to establish/determine who the tech lead for a particular feature is?

File: docs/reference/contributing/guidelines/workflow.md (hunk 12 above)
Comment (cmonr) [reply docs/reference/contributing/guidelines/workflow.md]: I'm not sure I follow either.

File: docs/reference/contributing/guidelines/workflow.md [hunk 15]
```diff
@@ -50,17 +51,21 @@ Pull requests on GitHub have to meet the following requirements to keep the code
...
 - Prefixing your commit message with a domain is acceptable, and we recommend doing so when it makes sense. However, prefixing one's domain with the name of the repo is not useful. For example, making a commit entitled "mbed-drivers: Fix doppelwidget frobulation" to the `mbed-drivers` repo is not acceptable because it is already understood that the commit applies to `mbed-drivers`. Renaming the commit to "doppelwidget: Fix frobulation" would be better, if we presume that "doppelwidget" is a meaningful domain for changes, because it communicates that the change applies to the doppelwidget area of `mbed-drivers`.
 - All new features and enhancements require documentation, tests and user guides for us to accept them. Please link each pull request to all relevant documentation and testing pull requests.
+- a pull request shall fix a bug, add a feature or refactor but not a mixture
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: Punctuation missing. Capitalization.

File: docs/reference/contributing/guidelines/workflow.md (hunk 5 above)
Comment (cmonr) [reply docs/reference/contributing/guidelines/workflow.md]: Numbers look fine when rendered: https://github.com/0xc0170/Handbook/blob/d65fe18cdb89888f01733f5c1b186830ea050dc1/docs/reference/contributing/guidelines/workflow.md

File: docs/reference/contributing/guidelines/workflow.md (hunk 6 above)
Comment (cmonr) [reply docs/reference/contributing/guidelines/workflow.md]: Fine when rendered.

File: docs/reference/contributing/guidelines/workflow.md [hunk 16]
```diff
@@ -80,14 +95,28 @@ Each pull request goes through the following workflow:
 
 Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
 
+Each state is time boxed. In most cases, this should be sufficient time to move to an another state.
+
 ##### Reviews
 
-All pull requests must be reviewed. The Mbed OS maintainers determine the most suitable person to review the pull request and tag that person accordingly. If a pull request requires a review from partners, the maintainers tag the corresponding GitHub group instead.
+All pull requests must be reviewed. The mbed CI bot determine the most suitable person to review the pull request and tag that person accordingly.
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: tags

File: docs/reference/contributing/guidelines/workflow.md [hunk 17]
```diff
@@ -80,14 +95,28 @@ Each pull request goes through the following workflow:
...
+
+##### Work needed
+
+The pull request requires additional work due to failed tests, changes requested.
+
+Time: 3 days to respond to the review comments
+
 ##### Releases
 
 Once we merge a pull request, we tag it with a release. This is the release in which we first publish this pull request. For patch releases, we allow only bug fixes, new targets and enhancements to existing functionality. New features only go to feature releases.
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: On line 122, any note about feature branch PRs not getting a release tag?

Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: will add

File: docs/reference/contributing/guidelines/workflow.md (hunk 16 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: fixed

File: docs/reference/contributing/guidelines/workflow.md (hunk 15 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: fixed

**Comment by 0xc0170** [comment]
I pushed few updates. All comments should be addressed. 

Restructured a bit the document, still missing something there, proposals ?

1. gatekeepers - the teams that are visible on github and users should know them
2. contributions
3. pull requests categories
4. pull requests states
5. additional labels

File: docs/reference/contributing/guidelines/workflow.md (hunk 14 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: we will have teams as we have for our partners

**Comment by cmonr** [comment]
Some overall thoughts on the document:
- Are we assuming that Github users have the same name/handle as in Mbed? (For CLA)
- In the Reporting Bugs section, I would suggest that it be made clear that users should submit instructions on how to reproduce the bug/issue, otherwise the bug will be closed.
- Do we have any guidance on stale issues?
- Do we have any guidance on user suggestions? Is it alright for users to open issue tickets for suggestions that may be large in scope? (Motivation: I'd like for us to _eventually_ get back to zero issues & PRs. Some of the issues currently open appear to be longstanding.)
- It might be useful to spell out CLA the first time it's used in the document.
- Replace _use rebase if you could_ with _always rebase when possible_
- _Pull requests shall fix a bug, add a feature or refactor but not a mixture._ I think there's suppose to be a comma here somewhere, but I'm not sure where.

**Comment by cmonr** [comment]
Oh. Something that dawned on me:
- What happens if a PR for a feature branch is opened by the feature branch's owner? Who is assigned for review?

**Comment by 0xc0170** [comment]
> Are we assuming that Github users have the same name/handle as in Mbed? (For CLA)

We don't. CLA will need more work (the bot, and even revise it we might)

> In the Reporting Bugs section, I would suggest that it be made clear that users should submit instructions on how to reproduce the bug/issue, otherwise the bug will be closed.

Will review

> Do we have any guidance on user suggestions? Is it alright for users to open issue tickets for suggestions that may be large in scope? (Motivation: I'd like for us to eventually get back to zero issues & PRs. Some of the issues currently open appear to be longstanding.)

This still needs to happen but outside of the scope for this (issue triage).

> It might be useful to spell out CLA the first time it's used in the document.

> Replace use rebase if you could with always rebase when possible

> Pull requests shall fix a bug, add a feature or refactor but not a mixture. I think there's suppose to be a comma here somewhere, but I'm not sure where.

Will fix

> What happens if a PR for a feature branch is opened by the feature branch's owner? Who is assigned for review?

Someone from the team (they should assign someone)

**Comment by 0xc0170** [comment]
@cmonr addressed all comments

**Pull request closed without merging**